### PR TITLE
Fix finetuning issue and improve training/inference workflow

### PR DESCRIPTION
**Pull request by CatoTea**
The finetuning issue was likely caused by an inappropriate learning rate distribution and decay strategy in the original repository. I have successfully finetuned a pre-trained model on my own animal dataset with good results after addressing this.

**Modifications:**
1. Replaced the learning rate scheduling strategy to decay based on validation loss.
2. Added checkpoint saving by epoch interval and best validation loss.
3. Refactored `inference.py` structure, adding support for folder input.
4. Updated `README.md` to include detailed finetuning instructions.

### CODE REVIEW

File: easy_ViTPose/configs/ViTPose_custom.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    dataset_channel=[
+        [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17],
+    ],
+    inference_channel=[
+        0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17
+    ])
+
+# Set models channels
+data_cfg['num_output_channels'] = channel_cfg['num_output_channels']
+data_cfg['num_joints']= channel_cfg['dataset_joints']
```
Comment (JunkyByte) [inline easy_ViTPose/configs/ViTPose_custom.py:16]: Could you fix formatting here? 👍

File: easy_ViTPose/vit_utils/train_valid_fn.py [hunk 2]
```diff
@@ -68,36 +68,22 @@ def train_model(model: nn.Module, datasets_train: Dataset, datasets_valid: Datas
...
-    
+    # Learning rate scheduler (MultiStepLR)
+    scheduler = ReduceLROnPlateau(optimizer, mode='min', factor=cfg.lr_config['factor'], patience=cfg.lr_config['patience'])
+   
     # AMP setting
     if cfg.use_amp:
         logger.info("Using Automatic Mixed Precision (AMP) training...")
         # Create a GradScaler object for FP16 training
-        scaler = GradScaler()
+        scaler = GradScaler(device='cuda')
```
Comment (JunkyByte) [inline easy_ViTPose/vit_utils/train_valid_fn.py:86]: Did training crash without specifying the device is cuda?

**Review by JunkyByte — CHANGES_REQUESTED** [review]

File: easy_ViTPose/vit_utils/train_valid_fn.py (hunk 2 above)
Comment (CatoTea) [reply easy_ViTPose/vit_utils/train_valid_fn.py:85]: Training didn’t crash. The previous usage torch.cuda.amp.GradScaler() is just deprecated, so I updated it.

**Comment by CatoTea** [comment]
I've pushed the changes. Thanks!

File: easy_ViTPose/configs/ViTPose_custom.py (hunk 1 above)
Comment (CatoTea) [reply easy_ViTPose/configs/ViTPose_custom.py]: fixed

**Comment by JunkyByte** [comment]
Thanks:)
